documentation/src/main/asciidoc/user_guide/server_protocols_hotrod_wire.adoc

```diff
@@ -2013,6 +2014,151 @@ Here are the current values for each flag: +
 |==============================================================================
 
 
+//TODO? increment version for transaction requests?
+
+
+
+
+.Prepare Request
+
+Request (0x3B):
+
+[cols="3,^2,10",options="header"]
+|===
+|Field Name |Size |Value
+
+|Header
+|variable
+|Request header
+
+|Xid
+|XID
+|The transaction ID (XID)
+
+|OnePhaseCommit
+|byte
+|When it is set to `1`, the server will use one-phase-commit if available (XA only)
+
+|Number of keys
+|vInt
+|The number of keys
+
+3+^|For each key
+
+| Key Length
+| vInt
+| Length of key.
+Note that the size of a vInt can be up to 5 bytes which in theory can produce bigger numbers than `Integer.MAX_VALUE`.
+However, Java cannot create a single array that’s bigger than `Integer.MAX_VALUE`, hence the protocol is limiting vInt array lengths to `Integer.MAX_VALUE`.
+
+|Key
+|byte array
+|Byte array containing the key
+
+|Version Read
+|long
+|The version read. `0` if `ReadState` is `NOT_READ` or `NON_EXISTING`.
+
+
+|Control Byte
+|Byte
+|The first 2 bits are the read status: +
++0x01+ = `NOT_READ` +
++0x02+ = `NON_EXISTING` +
+And the 3rd bit is set when it is a remove operation:
++0x04+ = `REMOVE_OPERATION` +
+
+|Lifespan
+|vInt
+|Number of seconds that a entry during which the entry is allowed to life.
+If number of seconds is bigger than 30 days, this number of seconds is treated as UNIX time and so, represents the number of seconds since 1/1/1970.
+If set to `0`, lifespan is unlimited.
+
+|Max Idle
+|vInt
+|Number of seconds that a entry can be idle before it’s evicted from the cache.
+If `0`, no max idle time.
+
+|Value Length
+|vInt
+|Length of value
+
+|Value
+|byte-array
+|Value to be stored
+
+|===
+
+.Commit and Rollback Request
+
+Request (0x3D and 0x3F):
+
+[cols="3,^2,10",options="header"]
+|===
+|Field Name |Size |Value
+
+|Header
+|variable
+|Request header
+
+|Xid
+|XID
+|The transaction ID (XID)
+
+|===
+
+.Response from prepare, commit and rollback request.
+
+Response 0x3C, 0x3E and 0x40
+
+[cols="3,^2,10",options="header"]
+|===
+|Field Name |Size |Value
+
+|Header
+|variable
+|Response header
+
+|XA return code
+|vInt
+|The XA code representing the prepare response. +
+Can be `XA_OK(0)`, `XA_RDONLY(3)` or any of the error codes (see `XaException`). +
+This field isn't present if the response state is different from `Successful`.
+
+|===
+
+
+.XID Formtat
+
+The XID in the requests has the following format:
```

Comment:
And that's part of the problem. But running a script within the scope of a transaction and not participating in that transaction is not correct. If you don't want to deal with that, clearly describe the limitations in docs.